mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -347,15 +385,24 @@ func resourceMongoDBAtlasSearchIndexCreate(ctx context.Context, d *schema.Resour
 	projectID := d.Get("project_id").(string)
 	clusterName := d.Get("cluster_name").(string)
 	dynamic := d.Get("mappings_dynamic").(bool)
+	var mappings *admin.ApiAtlasFTSMappings
+	indexType := d.Get("type").(string)
```

Comment:
looking only at this PR, I'm not sure where the default value is set ("search")
should we add this?
```suggestion
	indexType := d.Get("type").(string)
	if indexType == nil {
		indexType = "search" // move this to a const like vectorSearch
	}
```

Reply:
in TF if not stated explicitly the default values is the zero value, "" (empty string) in this case

Reply:
`If we change this for all resources they'll see change plans in search-index resources they didn't change: "" -> "search"`

Just to double-check, the already created indexes will always have type=""? Asking because, if the API starts returning type="search" for these old search indexes, TF will keep trying to reconcile the state and try to make type="" (as its old config says), or am I wrong?

e.g.:
2023-01-01 create search index without a type (not available at that moment)
2023-01-01 the GET API returns the search index without a type field
...
2023-11-15 the GET API starts returning the same search index with type="search" (instead of "") `// don't know if this will happen but it's important to check`
2023-11-16 TF detects a change (type in config is not set, so equals "", while the API says it's "search") even if you apply the plan (where the type is still "")

Reply:
the field doesn't exist in the state (terraform.tfstate) so when state is refreshed (e.g. before plan or apply) the state is updated to have:  type = ""    .   But this doesn't raise a plan change to the user.

if the SDK will start to return "search" then a plan change will be shown to the user.